sys/include/net/ng_udp.h

```diff
@@ -53,6 +53,13 @@ extern "C" {
 #endif
 
 /**
+ * @brief   The PID to the UDP thread.
+ *
+ * @note    Use @ref ng_udp_init() to initialize. **Do not set by hand**.
+ */
+extern kernel_pid_t ng_udp_pid;
```

Comment:
Yes: leave it private! Test are needed and a must have, but I don't like softening code because of tests! Especially in Black-box test environments like unittests, I say it (i) doesn't matter if the test code is larger/less efficient and (ii) you have full control over everything in test environments, there there won't be anyone else registering for NETTYPE_UDP in this situation!